### PR TITLE
YONK-805: Users view APIs for mobile app resource

### PR DESCRIPTION
**Pull request by aamishbaloch**
Changes include view for users view APIs for mobile app resource, through which users can be added or removed from the mobile apps. Unit tests are also included.

### CODE REVIEW

**Comment by aamishbaloch** [comment]
@ziafazal Can you review this?

File: mobileapps/views.py [hunk 1]
```diff
@@ -155,3 +167,57 @@ class MobileAppDetailView(SecureRetrieveUpdateAPIView):
...
+
+        GET /api/server/mobileapps/{id}/users
+        POST /api/server/mobileapps/{id}/users
+        DELETE /api/server/mobileapps/{id}/users
+
+        **POST Parameters**
+
+        The body of the POST request must include the following parameters.
+
+        * users_id: comma separated list of user ids to add into the mobile app
```
Comment (ziafazal) [inline mobileapps/views.py]: rename `user_id` should be `users`

Comment (ziafazal) [reply mobileapps/views.py]: We need to add docstring for DELETE method too.

File: mobileapps/views.py [hunk 2]
```diff
@@ -155,3 +167,57 @@ class MobileAppDetailView(SecureRetrieveUpdateAPIView):
...
+        by filtering against a 'mobileapp_id' in kwargs.
+        """
+        queryset = User.objects.filter(mobile_apps__exact=self.kwargs['mobileapp_id'])
+        return queryset
+
+    def post(self, request, mobileapp_id):
+        try:
+            mobileapp = MobileApp.objects.get(id=mobileapp_id)
+            user_ids = [int(user_id) for user_id in request.data['user_ids'].split(',')]
+            [mobileapp.users.add(user) for user in User.objects.filter(id__in=user_ids)]
```
Comment (ziafazal) [inline mobileapps/views.py]: what about simplifying this operation like this
```
for user in User.objects.filter(id__in=user_ids)
    mobileapp.users.add(user)
```

File: mobileapps/views.py [hunk 3]
```diff
@@ -155,3 +167,57 @@ class MobileAppDetailView(SecureRetrieveUpdateAPIView):
...
+        If the request is successful, the request returns an HTTP 200 "OK" response.
+    """
+    serializer_class = SimpleUserSerializer
+
+    def get_queryset(self):
+        """
+        Restricts the returned users to a given mobile app,
+        by filtering against a 'mobileapp_id' in kwargs.
+        """
+        queryset = User.objects.filter(mobile_apps__exact=self.kwargs['mobileapp_id'])
```
Comment (ziafazal) [inline mobileapps/views.py]: nit: just return here.

File: mobileapps/tests.py [hunk 4]
```diff
@@ -273,3 +269,84 @@ def test_mobileapps_inactive(self):
         self.assertEqual(response.status_code, 200)
         self.assertEqual(response.data['count'], 1)
         self.assertEqual(response.data['results'][0]['is_active'], False)
+
+    def test_mobileapps_users(self):
```
Comment (ziafazal) [inline mobileapps/tests.py]: We need to create a separate test Class for mobile apps users tests.

File: mobileapps/tests.py [hunk 5]
```diff
@@ -273,3 +269,84 @@ def test_mobileapps_inactive(self):
...
+        mobileapp = MobileApp.objects.create(
+            name='ABC App',
+            identifier='ABC',
+            operating_system=self.test_mobileapp_operating_system,
+            current_version=self.test_mobileapp_current_version,
+            updated_by=self.user,
+        )
+
+        users = UserFactory.create_batch(10)
+        mobileapp.users.add(*users)
```
Comment (ziafazal) [inline mobileapps/tests.py]: I think to make this test more close to real test case we should create at-least two mobile apps and distribute users among them.

File: mobileapps/views.py [hunk 6]
```diff
@@ -155,3 +167,57 @@ class MobileAppDetailView(SecureRetrieveUpdateAPIView):
...
+        return queryset
+
+    def post(self, request, mobileapp_id):
+        try:
+            mobileapp = MobileApp.objects.get(id=mobileapp_id)
+            user_ids = [int(user_id) for user_id in request.data['user_ids'].split(',')]
+            [mobileapp.users.add(user) for user in User.objects.filter(id__in=user_ids)]
+
+            return Response({}, status=status.HTTP_200_OK)
```
Comment (ziafazal) [inline mobileapps/views.py]: Http status code here should be 201.

File: mobileapps/views.py [hunk 7]
```diff
@@ -155,3 +167,57 @@ class MobileAppDetailView(SecureRetrieveUpdateAPIView):
...
+        except ObjectDoesNotExist:
+            raise Http404
+
+    def delete(self, request, mobileapp_id):
+        try:
+            mobileapp = MobileApp.objects.get(id=mobileapp_id)
+            user_ids = [int(user_id) for user_id in request.data['user_ids'].split(',')]
+            [mobileapp.users.remove(user) for user in User.objects.filter(id__in=user_ids)]
+
+            return Response({}, status=status.HTTP_200_OK)
```
Comment (ziafazal) [inline mobileapps/views.py]: Http status code here should be 204.

File: mobileapps/tests.py [hunk 8]
```diff
@@ -273,3 +269,84 @@ def test_mobileapps_inactive(self):
...
+        self.assertEqual(response.status_code, 200)
+
+        response = self.do_get(reverse('mobileapps-users', kwargs={'mobileapp_id': mobileapp.id}))
+        self.assertEqual(response.status_code, 200)
+        self.assertEqual(response.data['count'], len(users))
+        self.assertEqual(len(response.data['results']), len(users))
+        self.assertEqual(response.data['num_pages'], 1)
+
+    def test_mobileapps_users_remove(self):
+        mobileapp = MobileApp.objects.create(
```
Comment (ziafazal) [inline mobileapps/tests.py]: add a `setupClass` method of the test case class and add these operations into that
```
 mobileapp = MobileApp.objects.create(
            name='ABC App',
            identifier='ABC',
            operating_system=self.test_mobileapp_operating_system,
            current_version=self.test_mobileapp_current_version,
            updated_by=self.user,
        )

        users = UserFactory.create_batch(5)
        mobileapp.users.add(*users)
```

**Review by ziafazal — CHANGES_REQUESTED** [review]
@aamishbaloch completed first round of review.

**Comment by aamishbaloch** [comment]
@ziafazal Done with the suggested changes, Can you review the changes?

**Review by ziafazal — APPROVED** [review]
👍

**Pull request merged**